linera-core/src/updater.rs

```diff
@@ -97,14 +97,16 @@ pub enum CommunicationError<E: fmt::Debug> {
 
 /// Executes a sequence of actions in parallel for all validators.
 ///
-/// Tries to stop early when a quorum is reached. If `grace_period` is not zero, other validators
-/// are given this much additional time to contribute to the result, as a fraction of how long it
-/// took to reach the quorum.
+/// Tries to stop early when a quorum is reached. If `grace_period` is specified, other validators
+/// are given additional time to contribute to the result. The grace period is calculated as a fraction
+/// (defaulting to `DEFAULT_GRACE_PERIOD`) of the time taken to reach quorum.
 pub async fn communicate_with_quorum<'a, A, V, K, F, R, G>(
     validator_clients: &'a [RemoteNode<A>],
     committee: &Committee,
     group_by: G,
     execute: F,
+    // Grace period as a fraction of time taken to reach quorum
+    grace_period: impl Into<Option<f64>>,
```

Comment:
We generally avoid `impl Into` (to keep type-checking strong and the code normalized).

Reply:
reverted to `Option<f64>`